x/ccv/provider/keeper/power_shaping.go

```diff
@@ -0,0 +1,286 @@
+package keeper
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	storetypes "cosmossdk.io/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/interchain-security/v5/x/ccv/provider/types"
+	ccvtypes "github.com/cosmos/interchain-security/v5/x/ccv/types"
+)
+
+// GetConsumerPowerShapingParameters returns the power-shaping parameters associated with this consumer id
+func (k Keeper) GetConsumerPowerShapingParameters(ctx sdk.Context, consumerId string) (types.PowerShapingParameters, error) {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.ConsumerIdToPowerShapingParametersKey(consumerId))
+	if bz == nil {
+		return types.PowerShapingParameters{}, errorsmod.Wrapf(ccvtypes.ErrStoreKeyNotFound,
+			"GetConsumerPowerShapingParameters, consumerId(%s)", consumerId)
+	}
+	var record types.PowerShapingParameters
```

Comment:
Not part of this PR but since you "touch" this already, let's rename `record` (because it's not a record anymore) to `powerShapingParameters` or something similar.